### ISSUE TITLE
.all, toArray, toJSON not returning expected results

### ISSUE THREAD

**Issue by JCWolf** [ecrmnn/collect.js#138]
If I use those three methods(.all, toArray, toJSON) on a collection with sub-collections I get the `items` property from children in the result.

It shouldn't recursively get the items for each collection?

_Why I have sub-collections? Because I want, at some point, to process only the data from children and I don't want to create temporary variables and instantiate a new collection()._

**Comment by ecrmnn** [comment]
Yes you should.

Please submit a PR and I'll have a look :)

If you're not sure how to implement the code, submit only the tests that prove how it *should* work and we'll take it from there!

**Comment by Helfull** [comment]
Since this package wants to stay close to laravel collection, i guess we should take a look there

```
    $collect = collect([
        collect(['foo']),
        collect(['bar']),
    ]);
    dd($collect->all(), $collect->toJson(), $collect->toArray());
```
Gives me this in laravel
![image](https://user-images.githubusercontent.com/720645/32937921-b3929694-cb7a-11e7-8cf3-df35b047ac92.png)

the same code with collect.js gives me
![image](https://user-images.githubusercontent.com/720645/32937992-f9b8af1e-cb7a-11e7-8952-d2a48c5068c7.png)

so yeah there is clearly a problem with `toArray` and `toJson` within collectjs but `all` behaves like in laravel

The tests should be like this

```
  it('', function () {
    const firstChildCollection = collect(['foo']);
    const secondChildCollection = collect(['bar'])
    const collection = collect([firstChildCollection, secondChildCollection]);

    expect(collection.toArray()).to.eql([['foo'],['bar']]);
  });

  it('', function () {
    const firstChildCollection = collect(['foo']);
    const secondChildCollection = collect(['bar'])
    const collection = collect([firstChildCollection, secondChildCollection]);

    expect(collection.toJson()).to.eql('[["foo"],["bar"]]');
  });
```

right now i cant test it myself and make a PR maybe later the weekend

**Comment by ecrmnn** [comment]
I'm aware of it.

I haven't had the time to fix it. A PR would be really appreciated, @Helfull

**Comment by ecrmnn** [comment]
This is fixed in the ``4.0.0`` release.

**Comment by iiDestiny** [comment]
@ennetech Why does the result returned by `groupBy` have a items child element?
